ui/src/app/views/workspace/component.ts

```diff
@@ -283,6 +292,38 @@ export class WorkspaceComponent implements OnInit, OnDestroy {
     this.router.navigate(['share'], {relativeTo : this.route});
   }
 
+  sendANotebook(notebook): void {
+    let enableBtn = false;
+    let unSelectCheckBox = false;
+    this.checkColumnNotebook = true;
+    this.enablePushNotebookBtn = false;
+
+    for (const file of this.notebookList) {
+      if (file.push === false) {
+        this.checkColumnNotebook = false;
+        if (enableBtn) {
+          break;
+        }
+        unSelectCheckBox = true;
+      }
+      if (file.push === true) {
+        this.enablePushNotebookBtn = true;
+        if (unSelectCheckBox) {
+          break;
+        }
+        enableBtn = true;
+      }
+    }
+  }
+
+  sendAllNoteBooks(): void {
+    for (const file of this.notebookList) {
+      file.push = this.checkColumnNotebook;
```

Comment:
Not sure I like sticking extra properties onto this otherwise well-defined model; this is kind of breaking the nice properties of static typing in Typescript. Some kind of auxiliary map/array might be cleaner and more explicit. For example, I'm not really sure what this property means at the moment.